internal/cli/client.go

```diff
@@ -110,3 +137,20 @@ func WithToken(token string) Option {
 		pc.Client.SetAuthToken(token)
 	}
 }
+
+// WithRateLimitDisabled disables rate limiting
+func WithRateLimitDisabled() Option {
+	return func(pc *PortClient) {
+		pc.rateLimitManager.SetEnabled(false)
+	}
+}
+
+// WithRateLimitThreshold sets the threshold for when to start throttling
+// threshold should be between 0.0 and 1.0 (e.g., 0.1 means start throttling when 10% of requests remain)
+func WithRateLimitThreshold(threshold float64) Option {
+	return func(pc *PortClient) {
+		if threshold >= 0.0 && threshold <= 1.0 {
+			pc.rateLimitManager.SetThreshold(threshold)
+		}
+	}
+}
```

Comment:
This is too obscure. I think that you should return an error or panic if it is out of bounds.

Reply:
Another option is to `clamp` the threshold between `0` and `1`.